presto-main/src/main/java/com/facebook/presto/type/MultisetType.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.AbstractType;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+
+import java.util.List;
+
+import static com.facebook.presto.type.TypeUtils.parameterizedTypeName;
+import static java.util.Objects.requireNonNull;
+
+public class MultisetType
+        extends AbstractType
+{
+    private final Type elementType;
+    public static final String ARRAY_NULL_ELEMENT_MSG = "ARRAY comparison not supported for arrays with null elements";
+    // use an underlying arraytype for now, to delegate all calls to this.
+    private ArrayType delegate;
+
+    public MultisetType(Type elementType)
+    {
+        super(parameterizedTypeName("multiset", elementType.getTypeSignature()), Block.class);
+        this.elementType = requireNonNull(elementType, "elementType is null");
+        delegate = new ArrayType(elementType);
+    }
+
+    public Type getElementType()
+    {
+        return elementType;
+    }
+
+    @Override
+    public boolean isComparable()
```

Comment:
You need to implement `OperatorType.EQUAL` and `OperatorType.HASH_CODE` for any type that is comparable